tests/pgshovel/streams/states.py

```diff
@@ -203,11 +184,29 @@ def test_successful_transaction():
         {'commit_operation': commit},
     ]))
 
-    validated = validate(messages)
+    state = None
 
-    assert next(validated) == (InTransaction(messages[0].header.publisher, batch_identifier), messages[0])
-    assert next(validated) == (InTransaction(messages[1].header.publisher, batch_identifier), messages[1])
-    assert next(validated) == (Committed(messages[2].header.publisher, batch_identifier), messages[2])
+    state = reserialize(validate_transaction_state(state, 0, messages[0]))
+    assert get_oneof_value(state, 'state') == (
+        InTransaction(
+            publisher=messages[0].header.publisher,
+            batch_identifier=batch_identifier
+        )
+    )
+    state = reserialize(validate_transaction_state(state, 1, messages[1]))
+    assert get_oneof_value(state, 'state') == (
+        InTransaction(
+            publisher=messages[1].header.publisher,
+            batch_identifier=batch_identifier
+        )
+    )
+    state = reserialize(validate_transaction_state(state, 0, messages[2]))
```

Comment:
The offset here isn't increasing — which isn't necessarily a problem, but does read a little weird.